results_visualizations/create_map.py

```diff
@@ -0,0 +1,52 @@
+from typing import List, Dict
+
+import folium
+from folium.plugins import HeatMapWithTime
+
+
+def create_heated_map(stations: Dict, with_markers=False):
+    m = folium.Map(location=[51.0946, 17.026],
+                   tiles="Stamen Terrain",
+                   zoom_start=12)
+
+    df_hour_list = []
+    for hour in range(len(stations)):
+        stations_data = []
+        for loc in stations[hour].values():
+            stations_data.append(
+                [loc['lat'], loc['lng'], loc['rents']])
+        df_hour_list.append(stations_data)
+
+    HeatMapWithTime(df_hour_list,
+                    radius=8,
+                    gradient={0.2: 'blue', 0.4: 'lime', 0.6: 'orange',
+                              1: 'red'},
+                    min_opacity=0.8,
+                    max_opacity=1,
+                    use_local_extrema=True).add_to(m)
+
+    if with_markers:
+        for station, loc in stations[0].items():
+            folium.Marker(
+                [loc['lat'], loc['lng']],
+                popup=f'<i>{station}</i>'
+            ).add_to(m)
+
+    m.save('bikes_usage_map.html')
```

Comment:
fajnie by to było jakoś parametryzować aby aby odpalić dla kilku zbiorów jednocześnie. Ale nie jest to koniecznie

Reply:
Na razie odpalamy mamy tylko jeden zbiór, więc wystarczy, ale tak, przy większej ilości można by pomyśleć o jakimś automacie który zczyta dane z pliku i wygeneruje mape zapisujac do pliku dane_wejsciowe_map_{timestamp}.html i wtedy ta funkcje nalezałoby sparametryzować